dask/docs/design.md

```diff
@@ -0,0 +1,190 @@
+# Dask Cook Backend API
+
+## Motivation
+
+As part of our overall effort to introduce support for Dask workloads with GPU
+support in Kubernetes, we would like to be able to deploy jobs using the Dask
+API onto Cook. Although [`dask-kubernetes`](https://kubernetes.dask.org/en/latest)
+exists and some exploration has been done on that front, using Cook as a
+compute backend offers many benefits since it can federate access to multiple
+Kubernetes clusters.
+
+## Goals and Non-Goals
+
+This API aims to be as much of a "friction-free" environment for deploying Dask
+workloads on Cook as possible. As such, its primary goal is to be able to use
+the Cook backend in a "plug-and-play" fashion similar to other backends in
+Dask.
+
+Ideally, the process of using Cook on Dask should look something like using
+Kubernetes on Dask, as shown below (taken from [here](https://docs.dask.org/en/latest/setup/kubernetes.html)):
+
+```python
+from dask_kubernetes import KubeCluster
+cluster = KubeCluster.from_yaml('worker-template.yaml')
+cluster.scale(20)  # add 20 workers
+cluster.adapt()    # or create and destroy workers dynamically based on workload
+
+from dask.distributed import Client
+client = Client(cluster)
+```
```

Comment:
I'd suggest putting the CookCluster example here at the top, side by side with the KubeCluster example you have. I think it's okay that it's repeated at the bottom of the doc. Putting it up top gives readers a clear picture for what you want to build and they can compare the KubeCluster and proposed CookCluster APIs side by side.